examples/flutter_firebase_login/packages/authentication_repository/lib/src/authentication_repository.dart

```diff
@@ -3,18 +3,73 @@ import 'dart:async';
 import 'package:authentication_repository/authentication_repository.dart';
 import 'package:cache/cache.dart';
 import 'package:firebase_auth/firebase_auth.dart' as firebase_auth;
+import 'package:firebase_auth_platform_interface/firebase_auth_platform_interface.dart';
 import 'package:flutter/foundation.dart' show kIsWeb;
 import 'package:google_sign_in/google_sign_in.dart';
 import 'package:meta/meta.dart';
 
 /// Thrown if during the sign up process if a failure occurs.
-class SignUpFailure implements Exception {}
+class SignUpWithEmailAndPasswordFailure implements Exception {
+  SignUpWithEmailAndPasswordFailure(this.message);
+
+  final String message;
+
+  /// Create an authentication message from a firebase authentication exception code.
+  /// https://pub.dev/documentation/firebase_auth/latest/firebase_auth/FirebaseAuth/createUserWithEmailAndPassword.html
+  factory SignUpWithEmailAndPasswordFailure.fromCode(String code) {
+    switch(code) {
+      case 'invalid-email': return SignUpWithEmailAndPasswordFailure('Email is not valid or badly formatted.');
```

Comment:
hi @jasonjackson, can you format this:
```dart
case 'invalid-email': return SignUpWithEmailAndPasswordFailure('Email is not valid or badly formatted.');
```
to this:
```dart
case 'invalid-email': 
  return SignUpWithEmailAndPasswordFailure('Email is not valid or badly formatted.');
```
so it's more readable on each case? Let me know what you think. 👍